### DISCUSSION TITLE
Pydantic to Arrow Schema convertion support for Py Enum types

### DISCUSSION THREAD

**Discussion by janaka** [lancedb/lancedb#1836] [Q&A]
Hi, 
Is there a reason why Enum type fields are not supported when converting to schema (in _py_type_to_arrow_type()) ?

Something along the lines of the following throws the error `TypeError: Converting Pydantic type to Arrow Type: unsupported type <enum 'SomethingTypes'>`

I want the Pydantic model to be defined with "strong" types of course. I don't mind the value being stored as a string in LanceDB. 

I'm very new to using LanceDB and PyArrow. So want to understand reasons for currently not supporting Enums. As I see understand LanceDB tables are created from PyArrow schema, even if I give a Pydantic model under the hood `to_arrow_schema()` seems to be called and type mapping is handled by `_py_type_to_arrow_type()`. So I'm thinking of adding Enum support in my own code. I've not yet looked at data coming back out and hydrating the model correctly.

Is there a better / alternative to achieve my goal?

```py
class SomethingTypes(Enum):
  BIGTHING = "big thing"
  SMALLTHING = "small thing"

class MySomething(LanceModel):
  id: str
  name: str
  number_of_things: int
  something_type: SomethingTypes

something_arrow_schema = MySomething.to_arrow_schema()
table = self.index_store_context.dbconn.create_table(table_name, schema=something_arrow_schema)
```

**Comment by janaka** [comment]
Maybe this example from the docs has the answer. Maybe i can have a setter based on the Enum and the getter as return type str?

```py
from lancedb.pydantic import Vector, LanceModel

class Content(LanceModel):
    movie_id: int
    vector: Vector(128)
    genres: str
    title: str
    imdb_id: int

    @property
    def imdb_url(self) -> str:
        return f"https://www.imdb.com/title/tt{self.imdb_id}"

import pyarrow as pa
db = lancedb.connect("~/.lancedb")
table_name = "movielens_small"
table = db.create_table(table_name, schema=Content)
```

**Reply by QianZhu** [reply]
Hi @janaka thanks for raising up the issue! We need to have a fix from our side to support Pydantic Enum types conversion. For now, use the value type directly. Will keep you posted when the fix is in.
